.github/codeql/queries/assert-pure.ql

```diff
@@ -0,0 +1,29 @@
+/**
+ * @name Assert Pure
+ * @kind problem
+ * @problem.severity error
+ * @id vscode-codeql/assert-pure
+ * @description This query asserts that the modules tested in the pure-tests are "pure".
+ * This means that they don't have any transitive reference to vscode.
+ */
+import javascript
+
+class VSCodeImportDeclaration extends ImportDeclaration {
+  VSCodeImportDeclaration() {
+    this.getImportedPath().getValue() = "vscode"
+  }
+}
+
+ImportDeclaration transitiveImports(Module m) {
+  exists (ImportDeclaration i |
+    (result = i and i = m.getAnImport()) or
+    (result = transitiveImports(m.getAnImport().resolveImportedPath()))
+  )
+}
+
+from Module m, VSCodeImportDeclaration v
+where
+  // For now, let's use `src` and force this to fail. Should change to `pure`.
+  m.getFile().getParentContainer().getBaseName() = "pure" and
+  v = transitiveImports(m)
+select m.getFile(), "This module is not pure"
```

Comment:
The message will show up in the alert, so let's make it obvious what the problem is.
```suggestion
select m, "This module is not pure: it has a transitive dependency on the vscode API imported $@", v, "here"
```